acme_diags/driver/lat_lon_vector_driver.py

```diff
@@ -0,0 +1,197 @@
+                        elif var.find('U') ==0 or var.find('V') ==0:
+                            parameter.main_title = str(' '.join(['Wind', str(int(plev[ilev])), 'mb', season, region]))
+                        else:
+                            parameter.main_title = str(' '.join([var, str(int(plev[ilev])), 'mb', season, region]))
+    
+                        # Regrid towards the lower resolution of the two
+                        # variables for calculating the difference.
+                        mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
+                            mv1_domain, mv2_domain, parameter.regrid_tool, parameter.regrid_method)
+
+                        tests.append(mv1_reg)
+                        refs.append(mv2_reg)
+
+
+            # For variables without a z-axis.
+            else:
+                for region in regions:
+                    print("Selected region: {}".format(region))
+
+                    mv1_domain = utils.general.select_region(region, mv1, land_frac, ocean_frac, parameter)
+                    mv2_domain = utils.general.select_region(region, mv2, land_frac, ocean_frac, parameter)
+
+                    parameter.output_file = '-'.join(
+                        [ref_name, var_title, season, region])
+                    #parameter.main_title = str(' '.join([var, season, region]))
+                    if var.find('TAU') == 0:
+                        parameter.main_title = str(' '.join(['TAU', season, region]))
+                    elif var.find('U') == 0 or var.find('V') == 0:
+                        parameter.main_title = str(' '.join(['Near surface wind', season, region]))
+                    else:
+                        parameter.main_title = str(' '.join([var, season, region]))
+  
+
+                    # Regrid towards the lower resolution of the two
+                    # variables for calculating the difference.
+                    mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
+                        mv1_domain, mv2_domain, parameter.regrid_tool, parameter.regrid_method)
+
+                    tests.append(mv1_reg)
+                    refs.append(mv2_reg)
+
+        # Euclidean norm of the vector
+        test = (tests[ZONAL] **2.0 + tests[MERIDIONAL]**2.0)**0.5
+        ref = (refs[ZONAL] **2.0 + refs[MERIDIONAL]**2.0)**0.5
+        diffs = []
+        diff = test - ref
+        metrics_dict = create_metrics(ref, test, diff)
+        tests.append(test)
+        refs.append(ref)
+        diffs.append(tests[ZONAL] - ref[ZONAL])
+        diffs.append(tests[MERIDIONAL] - ref[MERIDIONAL])
+        # Note that diff is not the vector norm of the above two components.
+        # It is the difference of the test and ref vector norms.
+        diffs.append(diff)
+        #result = Results_Container(tests=tests, refs=refs, diff=diff, metrics = metrics_dict)
+
+        # Saving the metrics as a json.
```

Comment:
This code block appears to be sufficient to save the metrics dict to `json`. I removed the commented-out code blocks in the `if/else` for z-axis above.